src/handlers.rs

```diff
@@ -393,6 +399,84 @@ where
                 maybe_await!(wallet.verify_proof(&psbt, &msg, max_confirmation_height))?;
             Ok(json!({ "spendable": spendable }))
         }
+
+        // Payjoin Logic goes here
+        SendPayjoin{
+            bip21uri,
+        } => {
+            // convert the bip21 uri into a payjoin uri, and handle error if necessary 
+            let uri = payjoin::Uri::try_from(bip21uri)
+                .map_err(|e| Error::Payjoin(format!("Unable to convert BIP21 into Payjoin URI: {}", e)))?;
+            // ensure uri is payjoin capable
+            let uri = uri
+                .assume_checked()
```

Comment:
Rather than calling `assume_checked`, you want to use `require_network` and pass the configured wallet's network to ensure the string passed is a valid address you can spend to.

Reply:
You can call [`Wallet.network()`](https://docs.rs/bdk/latest/bdk/wallet/struct.Wallet.html#method.network) to get this data